### PR TITLE
fixes #41 by using a limited SizedQueue

### LINKED ISSUE #41: memory grows in producer
running this simple producer code will show that the memory of the process will increase continuously

```
require 'nsq'
producer = Nsq::Producer.new(
  nsqd: "10.0.0.1:4150",
  topic: "test#ephemeral"
)
i = 0
while $i < 30000000 do
  producer.write("simple nsq message with some bytes and text to show that memory grows up #$i")
  if $i%10000==0
    puts 'RAM USAGE: ' + `pmap #{Process.pid} | tail -1`[10,40].strip
  end
  $i +=1
end


print "finished producing...  sleeping 1min"
sleep(60)
producer.terminate

```

consumer:

```./nsq_tail --topic test#ephemeral --nsqd-tcp-address localhost:4150```

### PR DESCRIPTION
**Pull request by mschneider82**

### CODE REVIEW

File: lib/nsq/connection.rb [hunk 1]
```diff
@@ -55,7 +55,7 @@ def initialize(opts = {})
       end
 
       # for outgoing communication
-      @write_queue = Queue.new
+      @write_queue = SizedQueue.new(10000)
```
Comment (bschwartz) [inline lib/nsq/connection.rb]: @chen-anders Do you think this is a reasonable default (see context here: https://github.com/wistia/nsq-ruby/issues/41). Figured you'd have a good sense based on our internal operations and how bursty stuff can be.

Comment (chen-anders) [reply lib/nsq/connection.rb]: This seems reasonable though I wonder if there's a benefit to have this being configurable.

Comment (bschwartz) [reply lib/nsq/connection.rb]: @chen-anders and I just caught up about this. Feels fine to start with this as the default and maybe make it configurable if there's a case for it down the road. 👍

**Comment by bschwartz** [comment]
@mschneider82 PR looks great. Very elegant solution to this. Thanks for contributing. Going to add you to the README when I bump the version unless you're opposed :)

**Comment by bschwartz** [comment]
@mschneider82 Oh my bad! You're already listed as a contributor for your previous work. My apologies for forgetting about that!

**Comment by mschneider82** [comment]
NP thank you for merge!

**Pull request merged**